kubernetes/utils/operate_from_yaml.py

```diff
@@ -0,0 +1,229 @@
+                    k8s_client,
+                    yml_doc,
+                    operation,
+                    verbose,
+                    namespace=namespace,
+                    **kwargs)
+            except client.rest.ApiException as api_exception:
+                api_exceptions.append(api_exception)
+    else:
+        try:
+            operate_from_yaml_single_item(
+                k8s_client, yml_document, operation, verbose,
+                namespace=namespace, **kwargs
+            )
+        except client.rest.ApiException as api_exception:
+            api_exceptions.append(api_exception)
+
+    if api_exceptions:
+        raise FailToExecuteError(api_exceptions)
+
+
+def operate_from_yaml_single_item(
+        k8s_client,
+        yml_document,
+        operation,
+        verbose=False,
+        namespace="default",
+        **kwargs):
+    # get group and version from apiVersion
+    group, _, version = yml_document["apiVersion"].partition("/")
+    if version == "":
+        version = group
+        group = "core"
+    # Take care for the case e.g. api_type is "apiextensions.k8s.io"
+    group = "".join(group.rsplit(".k8s.io", 1))
+    # convert group name from DNS subdomain format to
+    # python class name convention
+    group = "".join(word.capitalize() for word in group.split('.'))
+    func = "{0}{1}Api".format(group, version.capitalize())
+    k8s_api = getattr(client, func)(k8s_client)
+    kind = yml_document["kind"]
+    kind = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', kind)
+    kind = re.sub('([a-z0-9])([A-Z])', r'\1_\2', kind).lower()
+    if operation == "create":
+        resp = create_k8s_object(
+            k8s_api, yml_document, kind, namespace=namespace)
+        if verbose:
+            msg = "{0} created.".format(kind)
+            if hasattr(resp, 'status'):
+                msg += " status='{0}'".format(str(resp.status))
+            print(msg)
+    elif operation == "delete":
+        resp = delete_k8s_object(
+            k8s_api, yml_document, kind, namespace=namespace)
+        if verbose:
+            msg = "{0} deleted.".format(kind)
+            if hasattr(resp, 'status'):
+                msg += " status='{0}'".format(str(resp.status))
+            print(msg)
+
+
+def create_k8s_object(k8s_api, yml_document, kind, **kwargs):
+
+    if hasattr(k8s_api, "create_namespaced_{0}".format(kind)):
+        if "namespace" in yml_document["metadata"]:
+            namespace = yml_document["metadata"]["namespace"]
+            kwargs['namespace'] = namespace
+        resp = getattr(k8s_api, "create_namespaced_{0}".format(kind))(
+            body=yml_document, **kwargs)
+    else:
+        kwargs.pop('namespace', None)
+        resp = getattr(k8s_api, "create_{0}".format(kind))(
+            body=yml_document, **kwargs)
+    return resp
+
+
+def delete_k8s_object(k8s_api, yml_document, kind, **kwargs):
+
+    if hasattr(k8s_api, "create_namespaced_{0}".format(kind)):
```

Comment:
Is the "create_namespaceD_{0}" here intentional? Should it not be "delete_namespaced_{0}"?